src/FieldArray.tsx

```diff
@@ -140,9 +140,9 @@ class FieldArrayInner<Values = {}> extends React.Component<
             : prevState.touched,
         };
       },
-      () => {
+      (state: FormikState<any>) => {
         if (validateOnChange) {
-          validateForm();
+          validateForm(state.values);
```

Comment:
Since `validateForm` is just an alias for `validateFormWithHighPriority`, there's a chance that `SET_ERRORS` won't be dispatched if validation results are equal to the current state. Any changes made using `setFormikState` will not have updated state yet.